processing/src/main/java/io/druid/query/topn/TopNMapFn.java

```diff
@@ -35,6 +37,8 @@ public TopNMapFn(
   {
     this.query = query;
     this.topNAlgorithm = topNAlgorithm;
+    this.dimension = query.getDimensionSpec().getDimension();
+    this.extractionFn = TopNQueryEngine.canApplyExtractionInPost(query) ? null : query.getDimensionSpec().getExtractionFn();
```

Comment:
I can tell you that the pre/post merge is only called on the broker currently, which is why the query rewrite works.